Solutions/STM32F4DISCOVERY/TinyBooter/TinyBooter.proj

```diff
@@ -23,6 +23,8 @@
         <ExtraTargets>$(ExtraTargets);CompressBin</ExtraTargets>
         <ScatterFileDefinition>scatterfile_bootloader_$(COMPILER_TOOL).$(SCATTER_EXT)</ScatterFileDefinition>
         <EXEScatterFileDefinition>scatterfile_bootloader_$(COMPILER_TOOL).$(SCATTER_EXT)</EXEScatterFileDefinition>
+        <!-- newlib-nano does not support long-long and long-double formatting, use standard newlib instead -->

```

Comment:
A fair point, though it presumes that all devices have some sort of firmware update support built into the silicon to allow updates even if the main application image is completely hosed.

Doing that for NETMF requires some major re-work of the wireProtocol and debugger communication layers (and MFDeploy). Effectively abstracting out the entire wire protocol so that a plug-in of some sort could manage all the actions needed by MFDeploy to update the NETMF image. Unfortunately most such facilities built into silicon and devices requires some sort of external hardware/jumper to engage the update. (Although for development purposes the abstraction could also allow for standard JTAG/SWD/CMSIS-DAP access too which would provide full access and control [Yeah, I've been thinking about this one a while!])